app/controllers/admin/move_controller.rb

```diff
@@ -3,44 +3,57 @@
 module Admin
   # Controller for moving resources between collections
   class MoveController < Admin::ApplicationController
-    before_action :set_work
-    before_action :set_status
-    before_action :set_work_form_from_cocina
-
     def new
       authorize!
 
-      @collections = Admin::MoveTargetCollections.call(work_form: @work_form)
+      @move_form = Admin::MoveForm.new(content_id: params[:content_id])
+      render :form
     end
 
-    def create
+    def create # rubocop:disable Metrics/AbcSize
       authorize!
 
-      collection = Collection.find_by(druid: params[:collection_druid])
-      Admin::Move.call(work_form: @work_form, work: @work, collection:, version_status: @version_status)
-
-      flash[:success] = I18n.t('messages.work_moved')
-      redirect_to wait_works_path(@work.id)
+      @move_form = Admin::MoveForm.new(work_form:, **move_params)
+
+      if @move_form.valid?
+        work_form.content_id = @move_form.content_id
+        Admin::Move.call(work_form:, work:, collection: @move_form.collection, version_status:)
+
+        flash[:success] = I18n.t('messages.work_moved')
+        # This breaks out of the turbo frame.
+        respond_to do |format|
+          format.turbo_stream do
+            render turbo_stream: turbo_stream.action(:full_page_redirect, wait_works_path(work.id))
+          end
+        end
```

Comment:
Would `redirect_to wait_works_path(work)` not break out of the frame here?

Reply:
No, it won't. This is definitely one of the turbo rough edges.